pymongo/encryption.py

```diff
@@ -144,7 +145,10 @@ def kms_request(self, kms_context):
             while kms_context.bytes_needed > 0:
                 # CSOT: update timeout.
                 conn.settimeout(max(_csot.clamp_remaining(_KMS_CONNECT_TIMEOUT), 0))
-                data = conn.recv(kms_context.bytes_needed)
+                try:
+                    data = conn.recv(kms_context.bytes_needed)
+                except Exception:
+                    data = conn.recv(kms_context.bytes_needed)
```

Comment:
The connection needs to be closed and recreated after a timeout. Also we can't retry after any Exception, only after transient errors.

Reply:
Oh interesting. closing/recreating won't work since we might have already called kms_context.feed() with a partial result. I'm not sure we can even do this until DRIVERS-1541 is implemented in libmongocrypt.

Reply:
I think that might be a wider scope than what is intended for this ticket. I think just retrying the WantRead error is good enough to reduce the incidence of test failures.

Reply:
Sure sounds fine to defer retry logic until [DRIVERS-1541](https://jira.mongodb.org/browse/DRIVERS-1541). However, this code is still flawed because it will exceed the CSOT timeout. It seems like you just want to double the default KMS socket_timeout from 10 seconds to 20 seconds. How about let's just do that explicitly rather adding any retry logic?